lib/spack/spack/bootstrap/environment.py

```diff
@@ -161,7 +161,7 @@ def _write_spack_yaml_file(self) -> None:
 
 def isort_root_spec() -> str:
     """Return the root spec used to bootstrap isort"""
-    return _root_spec("py-isort@4.3.5:")
+    return _root_spec("py-isort@5")
```

Comment:
```suggestion
    return _root_spec("py-isort@5:")
```
https://iscinumpy.dev/post/bound-version-constraints/

Reply:
This is minor, but I would still keep this. `isort` might have breaking changes on major versions, which is what was happening here. Also, we are not capping at the library level, but at the `spack style` dependency level.

Reply:
Up to you. I'm just afraid we'll forget to bump this to `5:6` once 6 is released. We'll eventually end up in the situation where someone is forced to use an old version of Spack (which only supports 5) but needs a newer isort for some reason due to a change in how isort works.